patterntest.py

```diff
@@ -1,55 +1,62 @@
-DISTANCE_THRESHOLD = 10  # units in feet, smallest distance of red ball as marker with wiggle room
-FRAME_WIDTH = 1920
-FRAME_HEIGHT = 1080
-LEFT_BOUND = 860  # 960 (center) - 100 px
-RIGHT_BOUND = 1060  # 960 (center) + 100 px
-
-
-def determinePattern(dist, x1, x2, x3):
-    center_ball_x = x1
-    if (x2 < RIGHT_BOUND and x2 > LEFT_BOUND):
-        center_ball_x = x2
-    elif (x3 is not None and x3 < RIGHT_BOUND and x3 > LEFT_BOUND):
-        center_ball_x = x3
-
-    left_zone_ball = False
-    right_zone_ball = False
-
-    if (x1 != center_ball_x and x1 < LEFT_BOUND):
-        left_zone_ball = True
-    elif (x1 != center_ball_x and x1 > RIGHT_BOUND):
-        right_zone_ball = True
-
-    if (x2 != center_ball_x and x2 < LEFT_BOUND):
-        left_zone_ball = True
-    elif (x2 != center_ball_x and x2 > RIGHT_BOUND):
-        right_zone_ball = True
-
-    if (x3 is not None and x3 != center_ball_x and x3 < LEFT_BOUND):
-        left_zone_ball = True
-    elif (x3 is not None and x3 != center_ball_x and x3 > RIGHT_BOUND):
-        right_zone_ball = True
-
-    print("left zone: ", left_zone_ball)
-    print("right zone: ", right_zone_ball)
-
-    pattern = patternGenerator(dist, left_zone_ball, right_zone_ball)
-    return pattern
-
-# Path A red: 1
-# Path A blue: 2
-# Path B red: 3
-# Path B blue: 4
-
-
-def patternGenerator(distance, left, right):
-    if(distance < DISTANCE_THRESHOLD):
-        if(left):
-            return 1
+DISTANCE_THRESHOLD = 14  # ft, distance to closest red balls w/ wiggle room
+PATHS = ['A RED', 'A BLUE', 'B RED', 'B BLUE']
+
+
+# dist is distance to closest ball, x_coords is array of the center coords of all balls detected
+def determinePattern(dist, x_coords):
+    path = getPathColor(dist)  # determine the path color
+
+    if len(x_coords) == 3:
+        x1 = x_coords[0]  # closest ball
+        x2 = x_coords[1]  # 2nd closest ball
+        x3 = x_coords[2]  # 3rd closest ball
+
+    if path:  # compare the red paths
+        # 180px = max deviation from x3 ball being aligned w/ x1 ball in path b red
+        if abs(x3-x1) > 180:
+            path = PATHS[0]  # path a red
         else:
-            return 3
-    else:
-        if(not right):
-            return 2
+            path = PATHS[2]  # path b red
+
+    else:  # compare the blue paths
+        if abs(x2-x1) > 450:  # subject to change, horizontal distance buffer (in px)
+            path = PATHS[1]  # path a blue
         else:
-            return 4
+            path = PATHS[3]  # path b blue
+
+    return path
+
+
+# returns if distance to closest ball is lower than threshold
+def getPathColor(distance):
+    return distance < DISTANCE_THRESHOLD  # true -> red, false -> blue
+
+# OLD STRATEGY
```

Comment:
We have git history to keep track of old approaches, you can just delete unused code.